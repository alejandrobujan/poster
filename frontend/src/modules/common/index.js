export {default as Pager} from "./components/Pager";
export {default as Errors} from "./components/Errors";
export {default as BackLink} from "./components/BackLink";
export {default as UserCard} from "./components/UserCard";
export {default as PostLink} from "./components/PostLink";
export {default as ImagesCard} from "./components/ImagesCard";