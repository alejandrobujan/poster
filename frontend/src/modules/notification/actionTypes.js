export const FIND_NOTIFICATIONS_COMPLETED = "poster/notifications/findNotificationsCompleted";
export const SET_VIEWED = "poster/notifications/setViewed";