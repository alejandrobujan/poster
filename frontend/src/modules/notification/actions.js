import * as actionTypes from './actionTypes';
import backend from '../../backend';

export const findNotifications = () => dispatch => {

    backend.notificationService.findNotifications(
        notifications => dispatch(findNotificationsCompleted(notifications)));

}
const findNotificationsCompleted = notifications => ({
    type: actionTypes.FIND_NOTIFICATIONS_COMPLETED,
    notifications
});