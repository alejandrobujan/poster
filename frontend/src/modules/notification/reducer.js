import { combineReducers } from 'redux';

import * as actionTypes from './actionTypes';

const initialState = {
	notifications: [],
};

const notifications = (state = initialState.notifications, action) => {

	switch (action.type) {

		case actionTypes.FIND_NOTIFICATIONS_COMPLETED:
			return action.notifications;

		case actionTypes.SET_VIEWED:
			return [...(state).map(n => (n.notificationId === action.notificationId ? { ...n, notificationViewed: true } : n))];

		default:
			return state;

	}

}


const reducer = combineReducers({
	notifications
});

export default reducer;