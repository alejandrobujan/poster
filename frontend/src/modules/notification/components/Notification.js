import { getDate } from '../../../backend/utils';
import { useDispatch } from 'react-redux';
import * as actions from '../actions';
import { useNavigate } from 'react-router-dom';

const Notification = ({ id, text, viewed, creationDate, postId, notifierUserDto }) => {
	
	const dispatch = useDispatch();
	const navigate = useNavigate();
	
	const handleNotificationClick = () => {
		dispatch(actions.markAsViewed(id));
		navigate(`/post/post-details/${postId}`);
	}

	return (
		<button data-testid="NotificationButton" className="dropdown-item border-bottom" style={{ overflowX: 'hidden' }} onClick={handleNotificationClick}>
			<div className='' style={{minWidth: '300px'}}>
				<div className="row text-center">
					<div className="col-2">
						<img data-testid="Avatar" src={notifierUserDto.avatar ? `data:image/*;base64,${notifierUserDto.avatar}` : "/poster/assets/profile.png"} alt="Avatar" height="30px" width="30px" />
					</div>
					<div className="col-8 text-center">
						<small style={{ maxWidth: '100%', display: 'block', whiteSpace: 'normal', fontSize: '12px' }}>
							<span data-testid="Date" style={{ color: 'grey', display: 'block', whiteSpace: 'nowrap' }}>{getDate(creationDate)}</span><br />
						</small>
					</div>
					<div className="col-2">
						{!viewed && <span data-testid="WarningNotification" className="badge bg-danger text-white rounded-pill"><small style={{color:'#dc3545'}}>&nbsp;â€¢</small></span>}
					</div>
				</div>
				<span data-testid="NotificationText" style={{ maxWidth: '100%', display: 'block', whiteSpace: 'normal', fontSize: '12px', textAlign: 'justify' }}>{text}</span>
			</div>
		</button>
	)
};

export default Notification;