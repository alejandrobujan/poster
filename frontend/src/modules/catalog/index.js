import * as actions from './actions';
import * as actionTypes from './actionTypes';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as SearchBar} from "./components/SearchBar";
export {default as CategorySelector} from './components/CategorySelector';
export {default as Post} from './components/Post';
export {default as Posts} from './components/Posts';
export {default as Feed} from './components/Feed';

// eslint-disable-next-line
export default {actions, actionTypes, reducer, selectors};


