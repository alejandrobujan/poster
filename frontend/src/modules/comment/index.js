import * as actions from './actions';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as Comments} from './components/Comments';
export {default as Comment} from './components/Comment';

// eslint-disable-next-line
export default  {actions, reducer, selectors};