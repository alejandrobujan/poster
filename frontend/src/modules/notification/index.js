import * as actions from './actions';
import * as actionTypes from './actionTypes';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as Notifications} from "./components/Notifications";
export {default as Notification} from "./components/Notification";

// eslint-disable-next-line
export default {actions, actionTypes, reducer, selectors};


