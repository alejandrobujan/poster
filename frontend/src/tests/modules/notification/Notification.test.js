import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";
import {Provider} from 'react-redux';

import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import renderer from "react-test-renderer";
import { notificationMock } from "../../state/Notification.mock";

import { Notification } from "../../../modules/notification";

describe("Notification", () => {
	
	const middlewares = [thunk];
    const mockStore = configureMockStore(middlewares);
    let store;
    store = mockStore(notificationMock);
    
    const notification = {
		id: 1,
		text: "Hola",
		viewed: false,
		creationDate: null,
		postId: 1,
		notifierUserDto: {
			avatar: null
		}
	}
    
    it("renders correctly", () => {
		const tree = renderer
        .create(
				<Provider store={store}>
					<MemoryRouter>
						<Notification id={notification.id} text={notification.text} viewed={notification.viewed} creationDate={notification.creationDate} postId={notification.postId} notifierUserDto={notification.notifierUserDto}/>
					</MemoryRouter>
				</Provider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
	
	 it("basic structure notification", async () => {
		render(
			<Provider store={store}>
				<MemoryRouter>
					<Notification id={notification.id} text={notification.text} viewed={notification.viewed} creationDate={notification.creationDate} postId={notification.postId} notifierUserDto={notification.notifierUserDto}/>
				</MemoryRouter>
			</Provider>
		);

		const notificationButton = screen.getByTestId('NotificationButton');

        const avatar = screen.getByTestId('Avatar');

        const date = screen.getByTestId('Date');
        
        const warningNotification = screen.getByTestId('WarningNotification');
        
        const notificationText = screen.getByTestId('NotificationText');

	});
	
	it("click notification button", async () => {
		render(
			<Provider store={store}>
				<MemoryRouter>
					<Notification id={notification.id} text={notification.text} viewed={notification.viewed} creationDate={notification.creationDate} postId={notification.postId} notifierUserDto={notification.notifierUserDto}/>
				</MemoryRouter>
			</Provider>
		);

		const notificationButton = screen.getByTestId('NotificationButton');

        fireEvent.click(notificationButton);

	});
    
});