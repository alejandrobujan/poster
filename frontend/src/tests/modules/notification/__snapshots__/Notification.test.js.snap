// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification renders correctly 1`] = `
<button
  className="dropdown-item border-bottom"
  data-testid="NotificationButton"
  onClick={[Function]}
  style={
    Object {
      "overflowX": "hidden",
    }
  }
>
  <div
    className=""
    style={
      Object {
        "minWidth": "300px",
      }
    }
  >
    <div
      className="row text-center"
    >
      <div
        className="col-2"
      >
        <img
          alt="Avatar"
          data-testid="Avatar"
          height="30px"
          src="/poster/assets/profile.png"
          width="30px"
        />
      </div>
      <div
        className="col-8 text-center"
      >
        <small
          style={
            Object {
              "display": "block",
              "fontSize": "12px",
              "maxWidth": "100%",
              "whiteSpace": "normal",
            }
          }
        >
          <span
            data-testid="Date"
            style={
              Object {
                "color": "grey",
                "display": "block",
                "whiteSpace": "nowrap",
              }
            }
          >
            01/01/1970 12:00:00 AM
          </span>
          <br />
        </small>
      </div>
      <div
        className="col-2"
      >
        <span
          className="badge bg-danger text-white rounded-pill"
          data-testid="WarningNotification"
        >
          <small
            style={
              Object {
                "color": "#dc3545",
              }
            }
          >
             •
          </small>
        </span>
      </div>
    </div>
    <span
      data-testid="NotificationText"
      style={
        Object {
          "display": "block",
          "fontSize": "12px",
          "maxWidth": "100%",
          "textAlign": "justify",
          "whiteSpace": "normal",
        }
      }
    >
      Hola
    </span>
  </div>
</button>
`;
